### ISSUE TITLE
The Edit votes page does not write voting changes if too many made at once

### ISSUE THREAD

**Issue by MichaelNMaggs** [hatnote/montage#197]
In a yes/no round, after making large numbers of changes to previous votes and hitting the Save Changes button, nothing happens. The Save Changes button still records the same number and the changes apepar not to be written to disk. Refreshing the page in Firefox causes all the edits to be lost.

Happens consistently but only when large numbers of voting changes have been made (over 100 changes definitely causes the issue)

MacOS 11.4 with Firefox 89.0.2

**Comment by userAdityaa** [comment]
Yes, When you try to save more than 100 vote changes at once, the backend rejects the request (it only allows up to 100 at a time), but the frontend doesn’t show any error and acts as if everything was saved. As a result, your changes are lost if you refresh.

**Comment by lgelauff** [comment]
The Edit Votes page (`VoteEdit.vue`) accumulates changes locally in an `edits` array. When the user clicks Save, all pending edits are sent in a single POST to `/v1/juror/round/<id>/tasks/submit`. The backend enforces a hard limit of 100 ratings per submission (`MAX_RATINGS_SUBMIT = 100` in `juror_endpoints.py`). If the limit is exceeded, `InvalidAction` is raised. The middleware catches this, returns HTTP 200 with `{"status": "exception", "errors": [...]}`. The frontend sees HTTP 200, `.catch()` never fires, and the save silently does nothing. The user has no indication their changes were not saved.

The Save button already shows the pending count (`Save (N)`), so the number is visible — but there is no warning before or after hitting 100.

Partial saves are not a risk: `edits` is only cleared on a successful `.then()`, so a prior successful save would remove those changes from the list before the next batch is sent.

## Reproduction steps

1. Log in as a juror in a yes/no or rating round with more than 100 images already voted on.
2. Navigate to Edit Votes.
3. Change the vote on more than 100 images without saving (the Save button shows the count climbing — no warning is shown at any point).
4. Click Save.
5. Observe: no error is shown, the button resets, and the page appears to have saved.
6. Refresh the page — all changes are gone.

## Two distinct problems

### Problem A — the user should be prompted to save well before hitting the backend limit

The 100-limit is a technical backstop; it is not a UX target. In practice, accumulating 25+ unsaved changes without saving is already risky (session expiry, accidental refresh). The frontend should nudge the user to save at a much lower threshold — around 25 pending edits — so the backend limit is never reached in normal use.

**Proposed approach:**

- At ~25 pending edits, show an inline prompt near the Save button: "You have 25 unsaved changes — consider saving now." Make this more aggressive at 50.
- The Save button already shows the count (`Save (N)`), so the prompt can be a simple colour or style change on that button, or a small inline warning beneath it.
- Optionally, disable further editing once pending edits get too close to the backend limit (100), with a message explaining why.

### Problem B — if the limit is hit anyway, the error must be surfaced and changes preserved

Even with a lower limit and frontend guards, the backend will still enforce the limit for other clients. The current silent-fail is a data-loss bug.

**Proposed approach:**

- **Frontend (short-term):** in `api.js`, reject the Promise when `response.data.status` is `'exception'` or `'failure'`. The `save()` call in `VoteEdit.vue` already has `.catch(alertService.error)` — it just never fires. This is the same interceptor fix tracked in #116.
- **Backend (potentially one-line):** return HTTP 400 for `InvalidAction` instead of HTTP 200. The class already extends `BadRequest` — the middleware just isn't using that information. Making the middleware propagate the HTTP status from `MontageError` subclasses would fix this category of silent failure at the source.
- **On edits preservation:** `edits.value` is only cleared inside `.then()` (confirmed in code), so a surfaced error will already leave the pending changes intact. No extra work needed here.

### Related: session expiry while edits are pending

If a session expires while the user has unsaved edits and they refresh to re-login, all pending changes are lost — Vue state is not persisted anywhere. Once #116 is fixed (auth failures surface a visible error), a user whose session expires mid-edit will see the error and can re-login in another tab without refreshing, then retry the save. Auto-saving (Problem A) would also reduce exposure by keeping the pending set small at all times.

**Comment by userAdityaa** [comment]
Alright @lgelauff, thanks for the guidance. I’ll push the updates based on your suggestions for #116 and then move on to this issue next.

I’ll be more mindful this time and make sure I follow your feedback closely.

**Comment by lgelauff** [comment]
Thanks! I didn't look at your PR yet on this, just wanted to start with the issue and making sure there's a clear definition on that as I realize that is not always clear.

**Comment by ayushshukla1807** [comment]
Traced this to `VoteEdit.vue` sending all edits in one API call. The backend enforces `MAX_RATINGS_SUBMIT = 100` in `juror_endpoints.py` and returns `400` for larger batches — so any save with >100 changes was silently rejected.

Fix in PR #496: chunks `edits.value` into sequential batches of 100 before submitting via a `Promise.reduce` chain. Jurors with any number of changes are now always fully saved.
